streams/src/main/java/org/apache/kafka/streams/kstream/internals/KTableNewProcessorSupplier.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.streams.kstream.internals;
+
+import org.apache.kafka.streams.processor.api.ProcessorSupplier;
+
+public interface KTableNewProcessorSupplier<KIn, VIn, KOut, VOut> extends ProcessorSupplier<KIn, Change<VIn>, KOut, Change<VOut>> {
```

Comment:
This is just a temporary class so that we don't have to migrate all the processors in a "big bang". Once the old processorSupplier is unused, we can rename this class back.

Reply:
Could we add TODO to the old interface to easily remind the potential removal work? Or we already have tickets to do it?

Reply:
I'll leave it to @jeqo to make sure this is done at the end of the migration. I don't think a TODO in the code is much use, but a subtask in @jeqo 's ticket would be effective.